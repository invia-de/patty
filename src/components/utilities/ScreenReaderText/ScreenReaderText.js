import React from 'react';
import PropTypes from 'prop-types';
import { screenreadertext } from './ScreenReaderText.module.scss';

export default function ScreenReaderText({ children, className, ...other }) {
  return (
    <div className={className} {...other}>
      {children}
    </div>
  );
}
ScreenReaderText.propTypes = {
  /** additional classNames you want to add */
  className: PropTypes.string,
  /** required due to accessibility */
  children: PropTypes.node.isRequired
};
ScreenReaderText.defaultProps = {
  className: screenreadertext
};
