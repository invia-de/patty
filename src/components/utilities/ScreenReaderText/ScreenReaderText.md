```jsx
<ScreenReaderText>some screen reader text</ScreenReaderText>
```
