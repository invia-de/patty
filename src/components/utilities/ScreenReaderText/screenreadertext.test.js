import React from 'react';
import ScreenReaderText from './ScreenReaderText';
import { render } from '@testing-library/react';

test('ScreenReaderText renders', () => {
  const { container } = render(
    <ScreenReaderText message="some message">
      <div>some children component</div>
    </ScreenReaderText>
  );
  expect(container.firstChild).toMatchSnapshot();
});
