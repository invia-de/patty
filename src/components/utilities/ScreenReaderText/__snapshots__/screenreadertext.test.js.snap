// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenReaderText renders 1`] = `
<div
  class="screenreadertext"
  message="some message"
>
  <div>
    some children component
  </div>
</div>
`;
